File: src/com/stephenclhc/BinarySearch.java
package com.stephenclhc;

public class BinarySearch {
    // 704 BinarySearch

    public int search(int[] nums, int target) {
        int leftIndex = 0;
        int rightIndex = nums.length - 1;

        while (leftIndex <= rightIndex) {
            int middleIndex = (leftIndex + rightIndex) / 2; // middleIndex = leftIndex + (leftIndex + rightIndex) / 2
            if(nums[middleIndex] == target) return middleIndex;
            if(nums[middleIndex] < target)
                leftIndex = middleIndex + 1;
            else rightIndex = middleIndex - 1;
        }
        return -1;
    }
}

File: Python/Binary_Search.py
#    704. BinarySearch

def search(self, nums: List[int], target: int) -> int:
    left_index = 0
    right_index = len(nums) - 1

    while left_index <= right_index:
        mid_index = left_index + (right_index - left_index) // 2
        if(nums[mid_index] == target):
            return mid_index
        if nums[mid_index] < target:
            left_index = mid_index + 1
        else:
            right_index = mid_index - 1

    return -1
